### DISCUSSION TITLE
Caching cdk.out in CI/CD

### DISCUSSION THREAD

**Discussion by devlsh** [aws/aws-cdk#16461] [Q&A]
I use GitHub Actions to deploy my stack based on folder changes, for example if `app` changes, run `cdk deploy app`.

This works great, however, since `cdk.out` doesn't exist (As it isn't part of the Git repo), it has to build all of the other stacks before deploying the stack I needed (i.e. any stacks with `Source.asset` that have build configs).

I can cache the `cdk.out` directory, however since it doesn't get pruned, it would continue to expand indefinitely.

Is there a better way around this?

**Comment by peterwoodworth** [comment]
I'm not entirely certain what your requirements are. If changes are being made to your app, you want your app to synthesize again, or else your `cdk.out` will be outdated. Maybe there's a better solution if you could elaborate on your requirements?
